docs/file-upload-with-graphql.md

```diff
@@ -0,0 +1,200 @@
+	resolver := &go_graphql_demo.Stub{}
+
+	resolver.MutationResolver.SingleUpload = func(ctx context.Context, file graphql.Upload) (*go_graphql_demo.File, error) {
+		content, err := ioutil.ReadAll(file.File)
+		if err != nil {
+			return nil, err
+		}
+		return &go_graphql_demo.File{
+			ID:      1,
+			Name:    file.Filename,
+			Content: string(content),
+		}, nil
+	}
+	resolver.MutationResolver.SingleUploadWithPayload = func(ctx context.Context, req go_graphql_demo.UploadFile) (*go_graphql_demo.File, error) {
+		content, err := ioutil.ReadAll(req.File.File)
+		if err != nil {
+			return nil, err
+		}
+		return &go_graphql_demo.File{
+			ID:      1,
+			Name:    req.File.Filename,
+			Content: string(content),
+		}, nil
+	}
+	resolver.MutationResolver.MultipleUpload = func(ctx context.Context, files []*graphql.Upload) ([]*go_graphql_demo.File, error) {
+		if len(files) == 0 {
+			return nil, errors.New("empty list")
+		}
+		var resp []*go_graphql_demo.File
+		for i := range files {
+			content, err := ioutil.ReadAll(files[i].File)
+			if err != nil {
+				return []*go_graphql_demo.File{}, err
+			}
+			resp = append(resp, &go_graphql_demo.File{
+				ID:      i + 1,
+				Name:    files[i].Filename,
+				Content: string(content),
+			})
+		}
+		return resp, nil
+	}
+	resolver.MutationResolver.MultipleUploadWithPayload = func(ctx context.Context, req []*go_graphql_demo.UploadFile) ([]*go_graphql_demo.File, error) {
+		if len(req) == 0 {
+			return nil, errors.New("empty list")
+		}
+		var resp []*go_graphql_demo.File
+		for i := range req {
+			content, err := ioutil.ReadAll(req[i].File.File)
+			if err != nil {
+				return []*go_graphql_demo.File{}, err
+			}
+			resp = append(resp, &go_graphql_demo.File{
+				ID:      i + 1,
+				Name:    req[i].File.Filename,
+				Content: string(content),
+			})
+		}
+		return resp, nil
+	}
+	return resolver
+}
+```
+
+
+</p>
+</details>
+
+<details>
+<summary> Resolver configuration </summary>
+<p>
+
+To make your server recognise that resolver, attach it inside `main` function. \
+You can also set some parameters like `UploadMaxMemory` or `UploadMaxSize`.
+
+```go
+resolver := getResolver()
+exec := fileupload.NewExecutableSchema(fileupload.Config{Resolvers: resolver})
+
+var mb int64 = 1 << 20
+uploadMaxMemory := handler.UploadMaxMemory(32 * mb)
+uploadMaxSize := handler.UploadMaxSize(50 * mb)
+
+http.Handle("/query", handler.GraphQL(exec, uploadMaxMemory, uploadMaxSize))
+```
+
+</p>
+</details>
+
+<details>
+<summary> Test request </summary>
+<p>
+
+The curl request accepts a `FILEPATH` variable with the path to file which you want to send with the request
+```bash
+curl localhost:8087/query \
+  -F operations='{ "query": "mutation ($file: Upload!) { singleUpload(file: $file) { id, name, content } }", "variables": { "file": null } }' \
+  -F map='{ "0": ["variables.file"] }' \
+  -F 0=@${FILEPATH}
+
+```
+
+</p>
+</details>
+
+For more examples check reference links section.
+
+#### Reference links
```

Comment:
I am wondering how this proposal can affect our API.
IMO we should provide flexible API for end-user that allows providing documentation content/API Spec/Event Spec:
a) in place, by specyfing it directly in mutation
b) by uploading a file, as it is presented in this proposal:

```
scalar Upload

input ClobInput {
    content: String
    file: Upload
}

type Mutation {
    storeDocumentation(in: ClobInput ): String
}

type Query {
    anything: String
}
```


Please update your proposal and add a new section with suggestions on how to change our graphql schema.